drivers/flash/Kconfig.nrf

```diff
@@ -27,6 +27,29 @@ config SOC_FLASH_NRF_RADIO_SYNC
 	help
 	  Enable synchronization between flash memory driver and radio.
 
+config SOC_FLASH_NRF_PARTIAL_ERASE
+	bool "Nordic nRFx flash driver partial erase"
+	depends on HAS_HW_NRF_NVMC_PE
```

Comment:
```suggestion
	depends on HAS_HW_NRF_NVMC_PE
	depends on SOC_FLASH_NRF_RADIO_SYNC
```

Reply:
Why? Cannot this feature be useful without synchronization with radio?

Reply:
Yeah, actually you have made that point last time.  It will work without sync. I am removing that line.

Reply:
... yeah I have even stated so below, in the description of the variable...

Reply:
> Why? Cannot this feature be useful without synchronization with radio?

I doubted in sens of use the feature without synchronization to radio.
 
But, maybe it can help to not lock the kernel scheduler for not so long as 90 ms.
This was not objective for this PR, but its additional gain.
Good point @anangl, thanks for force me to think about that.